src/main/webapp/app/admin/statistics/statistics.component.ts

```diff
@@ -0,0 +1,178 @@
+import { DataSet } from 'app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component';
+import { TranslateService } from '@ngx-translate/core';
+import * as moment from 'moment';
+
+export enum SpanType {
+    DAY = 'DAY',
+    WEEK = 'WEEK',
+    MONTH = 'MONTH',
+    YEAR = 'YEAR',
+}
+
+@Component({
+    selector: 'jhi-statistics',
+    templateUrl: './statistics.component.html',
+})
+export class JhiStatisticsComponent implements OnInit, OnChanges {
+    spanPattern = SPAN_PATTERN;
+    span: SpanType = SpanType.WEEK;
+
+    // Histogram related properties
+    public histogramData: number[] = [];
+    public barChartOptions: ChartOptions = {};
+    public barChartLabels: Label[] = [];
+    public barChartType: ChartType = 'bar';
+    public barChartLegend = true;
+    public UserLoginChartData: ChartDataSets[] = [];
+    public SubmissionsChartData: ChartDataSets[] = [];
+
+    @ViewChild(BaseChartDirective) chart: BaseChartDirective;
+
+    constructor(private service: StatisticsService, private translateService: TranslateService) {}
+
+    async ngOnInit() {
+        await this.setBinWidth();
+        await this.createChart();
+    }
+
+    async ngOnChanges() {}
+
+    private async setBinWidth(): Promise<void> {
+        switch (this.span) {
+            case SpanType.DAY:
+                this.histogramData = Array(24).fill(0);
+                break;
+            case SpanType.WEEK:
+                this.histogramData = Array(7).fill(0);
+                break;
+            case SpanType.MONTH:
+                const days = this.daysInMonth();
+                this.histogramData = Array(days).fill(0);
+                break;
+            case SpanType.YEAR:
+                this.histogramData = Array(12).fill(0);
+                break;
+        }
+    }
+
+    private daysInMonth(): number {
+        return new Date(new Date().getFullYear(), new Date().getMonth(), 0).getDate();
+    }
+
+    async getSubmissions(): Promise<number[]> {
+        return new Promise<number[]>((resolve, reject) => {
+            this.service.getTotalSubmissions(this.span).subscribe((res: number[]) => {
+                if (res !== null) {
+                    resolve(res);
+                } else {
+                    reject('Submissions could not get fetched');
+                }
+            });
+        });
+    }
+
+    onTabChanged(span: string): void {
+        switch (span) {
+            case 'Day':
+                this.span = SpanType.DAY;
+                break;
+            case 'Week':
+                this.span = SpanType.WEEK;
+                break;
+            case 'Month':
+                this.span = SpanType.MONTH;
+                break;
+            case 'Year':
+                this.span = SpanType.YEAR;
+                break;
+        }
+    }
+    private getWeekdays(day: number): string[] {
+        const days = [
+            this.translateService.instant('weekdays.monday'),
```

Comment:
we shouldn't introduce the labels for weekdays, moment can give them to you `momentObject.format('dddd')`

Reply:
[Done](https://github.com/ls1intum/Artemis/pull/2471/commits/499bb9fd70d8f4710fe3744189d0493f61a8db7d)

Reply:
@FuchsDominik did this not work, because the weekday translations are still there

Reply:
I implemented it as requested but withdrawed it again, as moment().format('dddd') does not translate the weekdays into german. I would handle it like this, unless you have another approach which would also translate the labels

Reply:
Resolved in [this] (https://github.com/ls1intum/Artemis/pull/2471/commits/f6e30d7546138808db3176d23669807099621dc0) commit